GenericNode/src/main/java/genericnode/Servers/TCPServer.java

```diff
@@ -0,0 +1,91 @@
+package genericnode.Servers;
+
+import genericnode.DataStorage;
+
+import javax.xml.crypto.Data;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.ArrayList;
+
+public class TCPServer extends Server {
+    ServerSocket tcpServer = null;
+    Socket connectionSocket = null;
+    DataInputStream inFromClient = null;
+    DataOutputStream outToClient = null;
+    @Override
+    public void startServer(int port) throws IOException {
+        tcpServer = new ServerSocket(port);
+        String operation = "";
+        while(!operation.equals("exit")) {
+            connectionSocket = tcpServer.accept();
+            inFromClient = new DataInputStream(connectionSocket.getInputStream());
+            outToClient = new DataOutputStream(connectionSocket.getOutputStream());
+            operation = inFromClient.readUTF();
+            executeOperation(operation);
+        }
+    }
+
+    @Override
+    public void put(String key, String value) throws IOException {
+        DataStorage.put(key, value);
+    }
+
+    @Override
+    public String get(String key) throws IOException {
+        String value = DataStorage.get(key);
+        outToClient.writeUTF(value);
```

Comment:
get returns not only value but also key like:
"get key=a get val=123"
Though I'm not sure the second "get" is needed...